Building on FreeBSD fails - gmt_common_sighandler.c

Originally posted in forum; https://forum.generic-mapping-tools.org/t/building-gmt-on-freebsd-12-1-release/967.

System:
```
$ uname -a
FreeBSD  12.1-RELEASE FreeBSD 12.1-RELEASE r354233 GENERIC  amd64
```
During ``cmake --build .``:
```
[... everything OK up until this point ...]
Scanning dependencies of target gmt
[100%] Building C object src/CMakeFiles/gmt.dir/gmt.c.o
[100%] Linking C executable gmt
ld: error: libgmt.so.6.2.0: undefined reference to backtrace
ld: error: libgmt.so.6.2.0: undefined reference to backtrace_symbols_fd
cc: error: linker command failed with exit code 1 (use -v to see invocation)
*** Error code 1

Stop.
make[2]: stopped in /usr/home/anbj/gmt/build
*** Error code 1

Stop.
make[1]: stopped in /usr/home/anbj/gmt/build
*** Error code 1

Stop.
make: stopped in /usr/home/anbj/gmt/build
```

Adding ``add_definitions(-DNO_SIGHANDLER)`` to *cmake/ConfigUser.cmake* makes it build successfully.

The error is likely caused by https://github.com/GenericMappingTools/gmt/blob/master/src/gmt_common_sighandler.c; the **#ifdefs** not being correct for FreeBSD.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -301,6 +301,11 @@ if (GMT_USE_THREADS AND NOT GLIB_FOUND)
 		CACHE INTERNAL "GTHREAD config message")
 endif (GMT_USE_THREADS AND NOT GLIB_FOUND)
 
+# FreeBSD uses a separate library for backtracking
+if (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
+	find_library (EXECINFO_LIBRARY NAMES execinfo)
+	list (APPEND GMT_OPTIONAL_LIBRARIES ${EXECINFO_LIBRARY})
+endif (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
 
 ##
 ##	Find pre-downloaded/pre-installed GSHHG and DCW data

